### PR TITLE
fix: update subhub validation call to match API

### LINKED ISSUE #1658: [fxa-auth-server] Ensure fxa-auth-server schemas for subhub calls are current
There is some drift on the validation of SubHub calls in fxa-auth-server, these should be resolved and current.

### PR DESCRIPTION
**Pull request by bbangert**
Closes #1658

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/validators.js [hunk 1]
```diff
@@ -293,11 +294,10 @@ module.exports.subscriptionsSubscriptionValidator = isA.object({
     .date()
     .timestamp('unix')
     .required(),
-  ended_at: isA.alternatives(
-    isA.date().timestamp('unix'),
-    isA.any().allow(null)
-  ),
-  nickname: isA.string().required(),
+  end_at: isA.alternatives(isA.date().timestamp('unix'), isA.any().allow(null)),
```
Comment (rfk) [inline packages/fxa-auth-server/lib/routes/validators.js]: Do we really want to allow `any` here, which IIUC just matches anything?

```
> end_at = isA.alternatives(isA.date().timestamp('unix'), isA.any().allow(null))
> end_at.validate({hello: "world"}).value 
{ hello: 'world' }
>
```

I *think* `isA.any().valid(null)` is what you want here instead; it's not any more intuitive but it seems to work in testing, allowing *only* `null` as the valid value.

File: packages/fxa-auth-server/lib/routes/validators.js [hunk 2]
```diff
@@ -293,11 +294,10 @@ module.exports.subscriptionsSubscriptionValidator = isA.object({
...
+  failure_code: isA.string().optional(),
+  failure_message: isA.string().optional(),
+  plan_name: isA.string().required(),
```
Comment (rfk) [inline packages/fxa-auth-server/lib/routes/validators.js]: Are we able to restrict the characters allowed in these strings, even if only to `routes.validators.DISPLAY_SAFE_UNICODE` or similar? I've seen too many bugs allowed through by the generic `.string()` validator...

Comment (bbangert) [reply packages/fxa-auth-server/lib/routes/validators.js]: Someday we can, yes.
